src/components/grid-list/grid-list.ts

```diff
@@ -1,8 +1,127 @@
-import {Component} from '@angular/core';
+import {
+  Component,
+  ViewEncapsulation,
+  AfterContentChecked,
+  OnInit,
+  Input,
+  ContentChildren,
+  QueryList,
+  Renderer,
+  ElementRef
+} from '@angular/core';
+import {MdGridTile} from './grid-tile';
+import {TileCoordinator} from './tile-coordinator';
+import {
+    TileStyler,
+    RowHeightMode,
+    FitTileStyler,
+    RatioTileStyler,
+    FixedTileStyler
+} from './tile-styler';
+import {Dir} from '../../core/rtl/dir';
+
+// TODO(kara): Conditional (responsive) column count / row size.
+// TODO(kara): Re-layout on window resize / media change (debounced).
+// TODO(kara): gridTileHeader and gridTileFooter.
 
 @Component({
   selector: 'md-grid-list',
+  host: { 'role': 'list' },
   templateUrl: './components/grid-list/grid-list.html',
   styleUrls: ['./components/grid-list/grid-list.css'],
+  encapsulation: ViewEncapsulation.None
 })
-export class MdGridList {}
+export class MdGridList implements OnInit, AfterContentChecked {
+  /** Number of columns being rendered. */
+  _cols: number;
+
+  /** Row height value passed in by user */
+  _rowHeight: string;
+
+  /** The amount of space between tiles. This will be something like '5px' or '2em'. */
+  _gutter: string = '1px';
+
+  /** Sets position and size styles for a tile */
+  _tileStyler: TileStyler;
+
+  /* Array of grid tiles being rendered (from query list) */
+  _tiles: MdGridTile[] = [];
+
+  constructor(private _renderer: Renderer, private _element: ElementRef,
+              private _dir: Dir) {}
+
+  get cols() { return this._cols; }
+
+  @Input('cols')
+  set cols(value: any) { this._cols = ensureNumberInput(value); }
+
+  get gutterSize() { return this._gutter; }
+
+  @Input('gutterSize')
+  set gutterSize(value: any) {
+    this._gutter = ensureStringInput(value);
+  }
+
+  /** Set internal representation of row height from the user-provided value. */
+  @Input('rowHeight')
+  set rowHeight(value: string | number) {
+    this._rowHeight = ensureStringInput(value);
+    if (value === RowHeightMode.FIT) {
+      this._tileStyler = new FitTileStyler();
+    } else if (this._rowHeight && this._rowHeight.match(/:/g)) {
+      this._tileStyler = new RatioTileStyler(this._rowHeight);
+    } else {
+      this._tileStyler = new FixedTileStyler(this._rowHeight);
+    }
+  }
+
+  /** Query list of tiles that are being rendered. */
+  @ContentChildren(MdGridTile)
+  set tiles(tiles: QueryList<MdGridTile>) {
+    this._tiles = tiles.toArray();
```

Comment:
It's part of the `tiles` setter, so it's called every time the `tiles` QueryList updates.  I would have kept the QueryList as it is, but it makes iteration in `layoutTiles` a pain because QueryList's  `forEach` doesn't support an index parameter.